workspaces/src/rpc/patch.rs

```diff
@@ -34,15 +37,21 @@ impl<'a, 'b> ImportContractBuilder<'a, 'b> {
             into_network,
             import_data: false,
             initial_balance: None,
+            block_id: None,
         }
     }
 
+    pub fn block_id(mut self, block_id: BlockId) -> Self {
+        self.block_id = Some(block_id);
+        self
+    }
```

Comment:
To abstract away this type, perhaps we can have `block_height` and `block_hash` be separate to avoid having to export the `BlockId` type as before? Unless we already need to expose the block height for the query outside of the builder pattern?

Reply:
interesting, but we'll need to export `CryptoHash` otherwise though. How about we just abstract away this type anyways so people don't have to import `BlockId` and only need to import `CryptoHash` if they need it for referencing specific block hashes, so that block height version doesn't need the import at all

Reply:
CryptoHash is just an alias, so less of an issue to import, but we could just have our own local type for this with no difference, as an option